src/main/java/team/asd/service/ProductService.java

```diff
@@ -1,31 +1,36 @@
 package team.asd.service;
 
 import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 import team.asd.dao.ProductDao;
+import team.asd.dto.ProductDto;
 import team.asd.entity.Product;
 import team.asd.exception.ValidationException;
+import team.asd.util.ConverterUtil;
 
+@Service
 public class ProductService {
 
     private final ProductDao productDao;
 
-    public ProductService(ProductDao productDao) {
+    public ProductService(@Autowired ProductDao productDao) {
         this.productDao = productDao;
     }
 
-    public Product readById(Integer id) {
+    public ProductDto readById(Integer id) {
         validId(id);
-        return productDao.readById(id);
+        return ConverterUtil.convertProductToDto(productDao.readById(id));
```

Comment:
@stishchenko please make conversion in controller. Let's service method returns entity object